### DISCUSSION TITLE
Mode shapes for 1st, 2nd, and 3rd critical speeds

### DISCUSSION THREAD

**Discussion by wigging** [petrobras/ross#960] [Q&A] [answered]
I am trying to use ROSS to get mode shapes similar to Figure 7 in [A Practical Guide to Rotor Dynamics](https://rotorbearingdynamics.com/wp-content/uploads/2018/04/A-Practical-Guide-to-Rotor-Dynamics.pdf). The mode shapes I'm trying to generate are on the right side of this figure:

<img width="709" alt="screenshot2" src="https://user-images.githubusercontent.com/6828967/226451818-aa4476c3-019b-4aba-ac88-a95316502a45.png">

This is for a solid steel shaft with a bearing at each end. The parameters I'm using for the model are:

```
Shaft dimensions
----------------
total length, L = 80 in (2.032 m)
diameter, D = 5 in (0.127 m)

Steel properties
----------------
Young's modulus, E = 211 GPa (211 x 10^9 N/m^2)
shear modulus, G = 81.2 GPa (81.2 x 10^9 N/m^2)
density, ρ = 7810 kg/m^3

Bearing stiffness range
-----------------------
min at kb = 10^3 lbf/in (1.75126 x 10^5 N/m)
max at kb = 10^6 lbf/in (1.75126 x 10^8 N/m)

Rotation speed range
--------------------
min at 100 RPM (10.47 rad/s)
max at 30,000 RPM (3141.59 rad/s)
```

My example code for a bearing stiffness of 10^6 lbf/in (1.75126 x 10^8 N/m) is shown below.

```python
import ross as rs

# Steel material properties
steel = rs.Material(name='Steel', rho=7810, E=211e9, G_s=81.2e9)

# Number of shaft elements
N_elem = 6

# Dimensions of a shaft element
L_total = 2.032
L = L_total / N_elem
i_d = 0
o_d = 0.127

shaft_elem = rs.ShaftElement(
    L=L,
    idl=i_d,
    odl=o_d,
    material=steel,
    shear_effects=True,
    rotary_inertia=True,
    gyroscopic=True
)

shaft_elements = [shaft_elem for _ in range(N_elem)]

# Bearings at beginning and end of shaft
bearing0 = rs.BearingElement(n=0, kxx=1.75126e8, cxx=0)
bearing1 = rs.BearingElement(n=N_elem, kxx=1.75126e8, cxx=0)
bearings = [bearing0, bearing1]

# Rotor assembly
rotor = rs.Rotor(shaft_elements=shaft_elements, bearing_elements=bearings)
fig = rotor.plot_rotor()
fig.show()

# Run modal analysis and store results
results = rotor.run_modal(speed=0)

# Plot 2D results using ROSS
fig = results.plot_mode_2d(mode=0, width=800, height=600)  # mode 1
fig.show()

fig = results.plot_mode_2d(mode=1, width=800, height=600)  # mode 2
fig.show()

fig = results.plot_mode_2d(mode=2, width=800, height=600)  # mode 3
fig.show()
```

The generated plots for the 1st, 2nd, and 3rd mode shapes are shown below. The 2nd and 3rd mode shapes are not similar to the shapes shown in Figure 7 from the reference mentioned above. Am I using the correct mode values, for example, is `mode=0` for the the 1st mode shape? Also, I set the `speed=0` but I'm not sure if that is correct for this problem. Any guidance on how to use ROSS for these mode shapes would be helpful. Thanks.

<img width="825" alt="screenshot" src="https://user-images.githubusercontent.com/6828967/226450965-2f33067e-90da-4ddf-81d3-71502f3d52e5.png">

**Comment by raphaeltimbo** [comment, marked as answer]
Hi @wigging !

 
> Am I using the correct mode values, for example, is `mode=0` for the the 1st mode shape? 

Yes, `mode=0` represents the first mode shape. However, note that in this scenario, `mode=0` and `mode=1` are quite similar, with mode=0 being the first **backward** mode and `mode=1` being the first **forward** mode. The terms backward and forward pertain to rotor whirl, which can occur in the same direction as the rotation (forward) or the opposite (backward). In the context of the undamped critical speed map you provided, discussions typically revolve around forward modes. As a result, the first plot on the right side of the figure corresponds to `mode=1`.

> Also, I set the `speed=0` but I'm not sure if that is correct for this problem. Any guidance on how to use ROSS for these mode shapes would be helpful. Thanks.

The speed in modal analysis impacts the results in two ways:

- By modifying the stiffness and damping coefficients for bearings and seals, which typically depend on speed/frequency. In your example, these coefficients are constant, so there is no influence in this case.
- By altering the gyroscopic effect. In your example, the gyroscopic effect's impact might be minimal, as your model does not contain any disks.

The gyroscopic effects can cause the backward and forward modes to 'split.' By adjusting the speed (e.g.`speed=1000`) and plotting `mode=0` and `mode=1` again, you will observe a minor difference in frequency.

As for `mode=3`, it corresponds to the second critical speed in your plot. By default, our plot utilizes the major axis, which is always positive. However, if we plot the values in the 'y' axis, for example, we obtain the following:

![image](https://user-images.githubusercontent.com/18506378/226462782-73c00a7a-d117-428f-b10b-04b4496a6ebe.png)

This option to plot the 'y' values is not exposed to the user for now. I will open an issue to implement that.

**Reply by wigging** [reply]
So how did you plot Y axis values if they are not available? Is it yn in the Shape object?

**Reply by wigging** [reply]
Also, what are the units associated with the Relative Displacement axis?

**Reply by wigging** [reply]
It looks like I can get to the y axis values with `results.shapes[3].plot_2d(orientation='y')`.

**Reply by raphaeltimbo** [reply]
Yes. Soon we will make the orientation argument available to the `plot_mode_2d` method.
Regarding the units, mode shapes relative displacements are dimensionless. 

If we run an unbalance response analysis, then we have a ‘deflected’ shape which will have ‘actual’ displacements.

For example, this code:
```python
import ross as rs
import numpy as np

rotor = rs.rotor_example()
frequency_range = np.linspace(315, 1150, 101)
results = rotor.run_unbalance_response(
    node=3,
    unbalance_magnitude=3,
    unbalance_phase=0,
    frequency=frequency_range,
)

results.plot_deflected_shape_3d(speed=frequency_range[4])

```
will produce this deflected shape:

![image](https://user-images.githubusercontent.com/18506378/226580363-27d5884c-0dfc-44a3-b4b5-ca76ef2c8aaf.png)

In this plot, we have ‘meter’ as the displacement unit.

This deflected shape is mainly influenced by:
 - Location of the forces (weight in this case);
 - Speed in which the rotor is running. Modes that are closer to the operational speed will have a larger impact on the deflected shape.

**Reply by wigging** [reply]
Ok, thanks again for the help. I look forward to the addition of the orientation argument to the `plot_mode_2d` method.
